dissect/target/plugins/os/unix/config.py

```diff
@@ -2,34 +2,47 @@
 
 import io
 import re
-from abc import abstractmethod
 from configparser import ConfigParser, MissingSectionHeaderError
+from dataclasses import dataclass
 from typing import Any, BinaryIO, ItemsView, Iterator, KeysView, Optional, TextIO, Union
 
 from dissect.target import Target
 from dissect.target.exceptions import ConfigurationParsingError
 from dissect.target.filesystem import Filesystem, FilesystemEntry, VirtualFilesystem
 from dissect.target.helpers import fsutil
+from dissect.target.plugin import Plugin, internal
 
 
 # TODO: Look if I can just create a parsing function and attach it to the
 # the parser below.
-class LinuxConfigurationParser:
-    def __init__(self, collapse: Optional[Union[bool, set]] = False) -> None:
+class ConfigurationParser:
+    def __init__(
+        self,
+        collapse: Optional[Union[bool, set]] = False,
+        seperator: tuple[str] = ("=",),
+        comment_prefixes: tuple[str] = (";", "#"),
+    ) -> None:
         self.collapse_all = collapse is True
         self.collapse = collapse if isinstance(collapse, set) else {}
+        self.seperator = seperator
+        self.comment_prefixes = comment_prefixes
         self.parsed_data = {}
 
-    @abstractmethod
     def parse_file(self, fh: TextIO) -> None:
-        ...
+        raise NotImplementedError()
 
     def __getitem__(self, item: Any) -> Union[dict, str]:
         return self.parsed_data[item]
 
     def __contains__(self, item: str) -> bool:
         return item in self.parsed_data
 
+    def get(self, item: str, default: Optional[Any] = None) -> Any:
+        try:
+            return self.parsed_data[item]
+        except KeyError:
+            return default
```

Comment:
If self.parsed_data is dict, why not do:

```suggestion
        return self.parsed_data.get(item, None)
```

?

Or do I miss something here?